### PR TITLE
Add Adobe Acrobat Reader DC

### PR DESCRIPTION
**Pull request by Reset816**
###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/winget-pkgs/pull/661)

### CODE REVIEW

**Comment by superusercode** [comment]
https://github.com/microsoft/winget-pkgs/pull/645#issuecomment-632226916 why not use the full installer that uses inno?

**Comment by superusercode** [comment]
@Reset12138 it is failing because the current script requires the user to interact with the installer outside of executing the install command. My version of the script allows you to avoid having to interact with that stub entirely.

**Comment by superusercode** [comment]
was a response to https://github.com/microsoft/winget-pkgs/pull/545#issuecomment-632390457 sorry

**Comment by Reset12138** [comment]
> [#645 (comment)](https://github.com/microsoft/winget-pkgs/pull/645#issuecomment-632226916) why not use the full installer that uses inno?

Is it correct to write down inno in InstallerType? I only found the exe, msi, msix are suppported.
image

**Comment by superusercode** [comment]
@Reset12138 In the spec inno is one of the installer types (see https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv0.1.md ) alongside nullsoft and wix. Can you link the site you got that template from so I can PR in additional information?

File: manifests/Adobe/AdobeAcrobatReaderDC/2020.009.20063.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+LicenseUrl: https://www.adobe.com/legal/terms.html
+Tags: PDF
+Description: View, sign, collaborate on, and annotate PDFs with our free Adobe Acrobat Reader. 
+Homepage: https://acrobat.adobe.com/us/en/acrobat/pdf-reader.html?promoid=TPQVLCXQ
+Installers:
+  - Arch: x86
+    Url: https://ardownload2.adobe.com/pub/adobe/reader/win/AcrobatDC/1901020064/AcroRdrDC1901020064_MUI.exe
+    Sha256: 81953F3CF426CBE9E6702D1AF7F727C59514C012D8D90BACFB012079C7DA6D23
+    InstallerType: exe
+    Language: Multilingual
```
Comment (superusercode) [inline manifests/Adobe/AdobeAcrobatReaderDC/2020.009.20063.yaml:15]:
```suggestion
Id: Adobe.AcrobatReaderDC
Version: 2020.009.20063
Name: Adobe Acrobat Reader DC
Publisher: Adobe
License: Copyright (c) Adobe
LicenseUrl: https://www.adobe.com/content/dam/cc/en/legal/licenses-terms/pdf/Reader-EULA-en_US-20181207.pdf
AppMoniker: adobereader
Description: Only with Adobe Acrobat Reader you can view, sign, collect and track feedback, and share PDFs for free. 
Homepage: https://acrobat.adobe.com/us/en/acrobat/pdf-reader.html
Installers:
  - Arch: x86
    Url: http://ardownload.adobe.com/pub/adobe/reader/win/AcrobatDC/2000920063/AcroRdrDC2000920063_en_US.exe
    Sha256: B49F6B49F28712735098836BB7D9BF2FF12568ED166936B7B90C87BC847B2C68
    InstallerType: inno
```

**Comment by Reset12138** [comment]
> @Reset12138 In the spec inno is one of the installer types (see https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv0.1.md ) alongside nullsoft and wix. Can you link the site you got that template from so I can PR in additional information?

template you mean where i get the installer? (sorry about my poor english)
I got the installer from [chocolatey](https://chocolatey.org/packages/adobereader) and fill other information with my own.

File: manifests/Adobe/AdobeAcrobatReaderDC/2020.009.20063.yaml (hunk 1 above)
Comment (Reset12138) [reply manifests/Adobe/AdobeAcrobatReaderDC/2020.009.20063.yaml:15]: I will go and check weather the AcroRdrDC2000920063_en_US.exe is the newest version.

**Comment by superusercode** [comment]
@Reset12138 sorry about not being clearer, I meant to ask what file the screenshot is taken from. I assume it's from the Microsoft Docs page for creating manifests?

File: manifests/Adobe/AdobeAcrobatReaderDC/2020.009.20063.yaml (hunk 1 above)
Comment (Reset12138) [reply manifests/Adobe/AdobeAcrobatReaderDC/2020.009.20063.yaml:15]: I found that AcroRdrDC2000920063_en_US.exe is based on Installshield not inno?
image
Or this something wrong with my understanding between Installshield and inno?

**Comment by Reset12138** [comment]
> @Reset12138 sorry about not being clearer, I meant to ask what file the screenshot is taken from. I assume it's from the Microsoft Docs page for creating manifests?

[here](https://github.com/microsoft/winget-pkgs#Authoring-a-Manifest)

**Comment by Reset12138** [comment]
> > @Reset12138 sorry about not being clearer, I meant to ask what file the screenshot is taken from. I assume it's from the Microsoft Docs page for creating manifests?
> 
> [here](https://github.com/microsoft/winget-pkgs#Authoring-a-Manifest)

It's seemed that

**Comment by Reset12138** [comment]
@superusercode the installer AcroRdrDC2000920063_en_US.exe only support en_US while AcroRdrDC1901020064_MUI.exe support multilingual. I just replace the installer in your commit.

**Comment by superusercode** [comment]
@Reset12138 I think you mixed the old installer link with the new installer type. The link I included in my suggestion downloads adobe reader before starting the installer rather than downloading it during.

**Comment by Reset12138** [comment]
> @Reset12138 I think you mixed the old installer link with the new installer type. The link I included in my suggestion downloads adobe reader before starting the installer rather than downloading it during.

Wait a minute. The [link ](https://ardownload2.adobe.com/pub/adobe/reader/win/AcrobatDC/1901020064/AcroRdrDC1901020064_MUI.exe)I given also downloads adobe reader before starting the installer

**Comment by superusercode** [comment]
well if it works then I think your link is the best way to do it since my link was region specific

**Review by NathanBnm — DISMISSED** [review]

File: manifests/Adobe/AdobeAcrobatReaderDC/2020.009.20063.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+Id: Adobe.AdobeAcrobatReaderDC
+Version: 2020.009.20063
+Name: Adobe Acrobat Reader DC
+Publisher: Adobe
+License: Copyright (c) Adobe
+LicenseUrl: https://www.adobe.com/content/dam/cc/en/legal/licenses-terms/pdf/Reader-EULA-en_US-20181207.pdf
+AppMoniker: adobereader
+Description: Only with Adobe Acrobat Reader you can view, sign, collect and track feedback, and share PDFs for free.
```
Comment (NathanBnm) [inline manifests/Adobe/AdobeAcrobatReaderDC/2020.009.20063.yaml:8]:
```suggestion
Description: The free global standard for reliably viewing, printing, signing, and commenting on PDF documents.
```

Comment (Reset12138) [reply manifests/Adobe/AdobeAcrobatReaderDC/2020.009.20063.yaml:8]: Jesus which one should I choose

Comment (NathanBnm) [reply manifests/Adobe/AdobeAcrobatReaderDC/2020.009.20063.yaml]: @Reset12138 I assume this is the most appropriate description given by Adobe. See this screenshot:
image

Comment (Reset12138) [reply manifests/Adobe/AdobeAcrobatReaderDC/2020.009.20063.yaml]: > @Reset12138 I assume this is the most appropriate description given by Adobe. See this screenshot:
> image

got it.
already merge yours

**Comment by Reset12138** [comment]
@superusercode Can you tell me how did you commit your code to my PR?

**Comment by Reset12138** [comment]
> @superusercode Can you tell me how did you commit your code to my PR?

image
I want to make some changes in other's PR.
what should i do？

**Comment by Reset12138** [comment]
> @superusercode Can you tell me how did you commit your code to my PR?

@NathanBnm do you have any idea about it？

**Comment by NathanBnm** [comment]
@Reset12138 you need to go to the changes tab of the wanted PR and here you can do some comments in the code with suggestions which can be applied by autjor

**Comment by NathanBnm** [comment]
@Reset12138 you can read more detailed info here https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request

**Comment by Reset12138** [comment]
> @Reset12138 you can read more detailed info here https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request

thank you very much

**Review by ghost — APPROVED** [review]
Validation has completed

**Comment by Reset12138** [comment]
great to work with you @superusercode @NathanBnm

**Comment by Reset12138** [comment]
@superusercode btw, how to know what the installer type is? how can i distinguish between inno and exe？
And the Adobe Acrobat Reader DC, i found it is InstallShild. Is InstallShild and inno are the same thing?
I can't find any document about it.
image

**Comment by superusercode** [comment]
@Reset12138 The easiest way to check right now is to use the ["Ultimate Silent Switch Finder"](https://deployhappiness.com/the-ultimate-exe-silent-switch-finder/), but it doesn't work consistently on newer versions of innosetup/nullsoft. Additionally, when the source for a program is available I try to search for nsis/iss in that repo to try and find which one it uses.

I don't think installshield is the same as innosetup, but it looks like the same flags worked if you were able to get this current manifest to install silently.

**Comment by Reset12138** [comment]
> @Reset12138 The easiest way to check right now is to use the ["Ultimate Silent Switch Finder"](https://deployhappiness.com/the-ultimate-exe-silent-switch-finder/), but it doesn't work consistently on newer versions of innosetup/nullsoft. Additionally, when the source for a program is available I try to search for nsis/iss in that repo to try and find which one it uses.
> 
> I don't think installshield is the same as innosetup, but it looks like the same flags worked if you were able to get this current manifest to install silently.

thank you that I learnt so much from your detailed reply. with the information you given i can do more and make this Project better together.

**Comment by SoftCreatR** [comment]
@Reset12138 Another possible way is to run an installer from command line and append `/?` to it. In most cases, a dialog will open, including all possible switches.

**Comment by Reset12138** [comment]
> @Reset12138 Another possible way is to run an installer from command line and append `/?` to it. In most cases, a dialog will open, including all possible switches.
> 
> In your specific case:
> 
> image

wow, first time to know such method. Learn so much from your comment. Thank you!

**Pull request merged**